src/consensus/validation.h

```diff
@@ -37,24 +37,21 @@ class CValidationState {
     std::string strDebugMessage;
 public:
     CValidationState() : mode(MODE_VALID), nDoS(0), chRejectCode(0), corruptionPossible(false) {}
-    bool DoS(int level, bool ret = false,
-             unsigned int chRejectCodeIn=0, const std::string &strRejectReasonIn="",
-             bool corruptionIn=false,
-             const std::string &strDebugMessageIn="") {
+    bool Invalid(int level, unsigned int chRejectCodeIn, const std::string& strRejectReasonIn, const std::string& strDebugMessageIn = "")
```

Comment:
I think you can make `strDebugMessageIn` a required argument for this and `Corrupt()` below, since all calls to both functions specify a `strDebugMessageIn`. That removes the final optional argument.

Reply:
It's interesting to consistently provide validation debug information though, that's a project I'm up for.